core/execution/common/market_activity_tracker_snapshot.go

```diff
@@ -656,6 +685,17 @@ func (mat *MarketActivityTracker) restore(tracker *snapshot.MarketTracker) {
 			mat.takerFeesPaidInEpoch = append(mat.takerFeesPaidInEpoch, epochMap)
 		}
 	}
+	if tracker.GameEligibilityTracker != nil {
+		for _, get := range tracker.GameEligibilityTracker {
+			mat.eligibilityInEpoch[get.GameId] = make([]map[string]struct{}, 0, len(get.EpochEligibility))
```

Comment:
```suggestion
			mat.eligibilityInEpoch[get.GameId] = make([]map[string]struct{}, len(get.EpochEligibility))
```

Same as above comment, to stop the panic.